internal/storage/memory.go

```diff
@@ -267,13 +269,44 @@ func (m *MemoryConnector) DeleteLogs(logs []common.Log) error {
 }
 
 func (m *MemoryConnector) InsertBlockData(data []common.BlockData) error {
+	for _, blockData := range data {
+		dataJson, err := json.Marshal(blockData)
+		if err != nil {
+			return err
+		}
+		m.cache.Add(fmt.Sprintf("blockData:%s", blockData.Block.Number.String()), string(dataJson))
+	}
 	return nil
 }
 
-func (m *MemoryConnector) GetBlockData(blockNumbers []*big.Int) (data []common.BlockData, err error) {
-	return nil, nil
+func (m *MemoryConnector) GetBlockData(qf QueryFilter) ([]common.BlockData, error) {
+	blockData := []common.BlockData{}
+	limit := getLimit(qf)
+	blockNumbersToCheck := getBlockNumbersToCheck(qf)
+
+	for _, key := range m.cache.Keys() {
+		if len(blockData) >= int(limit) {
+			break
```

Comment:
shouldn't we log an error here to know that large blockdata is being requested?

Reply:
no, this just means that the requested limit is filled and we should return the data scanned thus far